src/Network/WebSockets/TSSocketManager.m

```diff
@@ -110,9 +105,9 @@ - (void)becomeActive
     }
 
     // Discard the old socket which is already closed or is closing.
-    [socket close];
-    self.status = kSocketStatusClosed;
-    socket.delegate = nil;
+    [self closeWebSocket];
+
+    self.status = kSocketStatusConnecting;
```

Comment:
This was a bug (but not a regression) - we weren't immediately updating status when we created a new websocket.